### PR TITLE
Fix date formatting timezone, replace react-native-locale by react-native-localize

### PR DESCRIPTION
**Pull request by OlivierFreyssinet**
### The issue:
All the dates in the app were formatted in UTC timezone.

The issue was that since we were using Hermes as a JS engine, we had to polyfill DateTimeFormat with FormatJS, which came with UTC set a default and left us with **no "javascript native" way to get the current timezone.**

### Changes:
- Replacing `react-native-locale` (old 4y outdated lib) by `react-native-localize` (more modern, **gives us access to the system timezone**).
- Adapting to the change of library:
  - **`src/languages.js`**:
     The default app language can now be obtained directly from `react-native-localize` with `findBestAvailableLanguage(listOfSupportedLanguages)`.
  - **`src/analytics/segment.js`**:
     In the analytics global properties:
    - removing property `localeIdentifier` which was irrelevant on iOS
    - removing property `preferredLanguage` which was working only on iOS
    - adding a property `systemLanguage` which has better naming, works on iOS and Android and gives us the system's main language, fulfilling the role of what the two removed properties were doing before.
- Fixing the DateTimeFormat timezone issue: 
  - **`src/context/Locale.js`**:
    Using `react-native-localize`'s `getTimeZone()` and using it as a default timezone for `DateTimeFormat`.

### Result:
How I tested it: example in timezone Europe/Paris using `<FormatDate date={new Date()} withHoursMinutes />`
> Screenshot_1649411357

### Type

Bug fix

### Context

v3 prerelease

### Parts of the app affected / Test plan

App's default language.
Analytics.
All displayed dates.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -155,7 +155,7 @@
     "react-native-gesture-handler": "^1.10.3",
     "react-native-keychain": "^7.0.0",
     "react-native-level-fs": "^3.0.0",
-    "react-native-locale": "0.0.19",
+    "react-native-localize": "^2.2.1",
```
Comment (ofreyssinet-ledger) [inline package.json:158]: - Replacing `react-native-locale` (old 4y outdated lib) by `react-native-localize` (more modern, **gives us access to the system timezone**).

File: src/analytics/segment.js [hunk 2]
```diff
@@ -52,8 +53,7 @@ const extraProperties = store => {
     androidVersionCode: getAndroidVersionCode(VersionNumber.buildVersion),
     androidArchitecture: getAndroidArchitecture(VersionNumber.buildVersion),
     environment: ANALYTICS_LOGS ? "development" : "production",
-    localeIdentifier,
-    preferredLanguage: preferredLanguages ? preferredLanguages[0] : null,
+    systemLanguage,
```
Comment (ofreyssinet-ledger) [inline src/analytics/segment.js:56]: Adapting to the change of library:
- **`src/analytics/segment.js`**:
     In the analytics global properties:
    - removing property `localeIdentifier` which was irrelevant on iOS
    - removing property `preferredLanguage` which was working only on iOS
    - adding a property `systemLanguage` which has better naming, works on iOS

File: src/languages.js [hunk 3]
```diff
@@ -57,17 +57,9 @@ export const DEFAULT_LANGUAGE_LOCALE = "en";
...
-  const matches = locale.match(/([a-z]{2,4}[-_]([A-Z]{2,4}|[0-9]{3}))/);
-  const lang = (matches && matches[1].replace("_", "-")) || "en-US";
-  return (
-    fullySupportedLocales.find(locale => lang.startsWith(locale)) ||
-    fallbackLocale
-  );
-};
+) =>
+  RNLocalize.findBestAvailableLanguage(fullySupportedLocales)?.languageTag ||
+  fallbackLocale;
```
Comment (ofreyssinet-ledger) [inline src/languages.js:62]: - Adapting to the change of library:
 - **`src/languages.js`**:
     The default app language can now be obtained directly from `react-native-localize` with `findBestAvailableLanguage(listOfSupportedLanguages)`.

File: src/context/Locale.js [hunk 4]
```diff
@@ -18,6 +19,12 @@ import {
 } from "../languages";
 import { languageSelector } from "../reducers/settings";
 
+if ("__setDefaultTimeZone" in Intl.DateTimeFormat) {
+  /** https://formatjs.io/docs/polyfills/intl-datetimeformat/#default-timezone */
+  // $FlowFixMe
+  Intl.DateTimeFormat.__setDefaultTimeZone(getTimeZone()); // eslint-disable-line no-underscore-dangle
+}
+
```
Comment (ofreyssinet-ledger) [inline src/context/Locale.js:27]: Fixing the DateTimeFormat timezone issue: 
  - **`src/context/Locale.js`**:
    Using `react-native-localize`'s `getTimeZone()` and using it as a default timezone for `DateTimeFormat`.

**Review by LFBarreto — APPROVED** [review]
This PR does spark joy :sparkle:

**Pull request merged**